### PR TITLE
Excel upload + View Table uploaded Students

### PR DESCRIPTION
**Pull request by Kutatsu**

### CODE REVIEW

File: app/Http/Controllers/StudentController.php
Comment (Bacio001-CCG) [inline app/Http/Controllers/StudentController.php:1]: Even though \Maatwebsite\Excel\Validators\ValidationException is a valid way of importing, I would suggest doing it at the top of the file and just referencing the part you need

File: app/Imports/UsersImport.php
Comment (Bacio001-CCG) [inline app/Imports/UsersImport.php:1]: Errors are in english but they need to be displayed in dutch

File: app/Http/Controllers/StudentController.php
Comment (Bacio001-CCG) [inline app/Http/Controllers/StudentController.php:1]: Remove the comments if not used

File: composer.lock
Comment (Bacio001-CCG) [inline composer.lock:1]: This is not supposed to come with it mind putting it in the .ignore if it isn't already

File: database/factories/DeanFactory.php
Comment (Bacio001-CCG) [inline database/factories/DeanFactory.php:1]: Make sure that these factories don't break the factories from development up on merging pls

File: package-lock.json
Comment (Bacio001-CCG) [inline package-lock.json:1]: This also should be in the .gitignore

File: resources/views/users.blade.php
Comment (Bacio001-CCG) [inline resources/views/users.blade.php:1]: Not really sure what this is supposed to be but I don't think you want an empty body with just a script tag. Please keep import and script stuff in the layout files, also use our layout in general.

**Review by Bacio001-CCG — CHANGES_REQUESTED** [review]
Please fix the requests before merging.

File: app/Http/Controllers/StudentController.php [hunk 1]
```diff
@@ -27,7 +29,28 @@ public function create()
...
+        } catch (\Maatwebsite\Excel\Validators\ValidationException $e) {
+             $failures = $e->failures();
+             return redirect()->back()->with('import_errors', $failures);
+             
+            //  foreach ($failures as $failure) {
+            //      $failure->row(); // row that went wrong
+            //      $failure->attribute(); // either heading key (if using heading row concern) or column index
+            //      $failure->errors(); // Actual error messages from Laravel validator
+            //      $failure->values(); // The values of the row that has failed.
+            //  }
```
Comment (MikeWhileCoding) [inline app/Http/Controllers/StudentController.php:48]:
```suggestion
```

Remove commented code if not needed

File: database/factories/DeanFactory.php
Comment (MikeWhileCoding) [inline database/factories/DeanFactory.php:1]: Make sure to merge factory from dev unless there is a big difference, then merge it right

File: resources/views/users.blade.php
Comment (MikeWhileCoding) [inline resources/views/users.blade.php:1]: No content except for header? Is there a future need for this?

File: routes/web.php
Comment (MikeWhileCoding) [inline routes/web.php:1]: Look at the new way of the route grouping

File: tests/Unit/StoreFunctionUnitTest.php
Comment (MikeWhileCoding) [inline tests/Unit/StoreFunctionUnitTest.php:1]: Do these test work? Naming is different from the already existing tests in dev

File: tests/Unit/UsersImportUnitTest.php
Comment (MikeWhileCoding) [inline tests/Unit/UsersImportUnitTest.php:1]: Same as other test

**Review by MikeWhileCoding — CHANGES_REQUESTED** [review]

File: tests/Unit/StoreFunctionUnitTest.php
Comment (Bacio001-CCG) [reply tests/Unit/StoreFunctionUnitTest.php:1]: These probably don't run naming convention are wrong I'm 99% sure

**Comment by Kutatsu** [comment]
@ITZZwilco Needs to look at the factory issue, because i didn't touch that so no idea what happened there.

**Comment by MikeWhileCoding** [comment]
> @ITZZwilco Needs to look at the factory issue, because i didn't touch that so no idea what happened there.

I know what happened here 😄 You added the school_id. I will check and fix the merge conflics in minute

**Review by Bacio001-CCG — APPROVED** [review]

**Review by MikeWhileCoding — APPROVED** [review]

**Pull request merged**